dags/sql_examples/sql_check_redshift_etl.py

```diff
@@ -0,0 +1,244 @@
+from airflow.decorators import task
+
+import pandas as pd
+
+
+DATES = ["2019-01", "2019-02"]
+TASK_DICT = {}
+
+with DAG("sql_data_quality",
+         start_date=datetime(2021, 7, 7),
+         description="A sample Airflow DAG to perform data quality checks using SQL Operators.",
+         schedule_interval=None,
+         default_args={"conn_id": "redshift_default"},
+         catchup=False) as dag:
+    """
+    ### SQL Check Operators Data Quality ETL Example
+
+    Before running the DAG, set the following in an Airflow or Environment Variable:
+    - key: aws_configs
+    - value: { "s3_bucket": [bucket_name], "s3_key_prefix": [key_prefix], "redshift_table": [table_name]}
+    Fully replacing [bucket_name], [key_prefix], and [table_name].
+
+    See the README for information on how to set up your Redshift connection.
+    This DAG can be used with other databases as long as the Redshift (and possibly
+    transfer operators) are changed.
+    """
+
+    begin = DummyOperator(task_id="begin")
+    end = DummyOperator(task_id="end")
+    converge_1 = DummyOperator(task_id="converge_1")
+    converge_2 = DummyOperator(task_id="converge_2")
+
+    @task
+    def add_upload_date(file_path, upload_date):
+        """
+        #### Transform Task
+        In general, it is not recommended to perform transform operations in
+        Airflow Tasks, as Airflow is designed to be an orchestrator, not a
+        computation engine. However, the transform is done here as it is a
+        relatively small operation, simply adding an upload_date column to the
+        dataframe for use in the SQL data quality checks later. Doing the
+        transform here also makes this example more easily extensible to the
+        use of other backend datastores.
+        """
+        trip_dict = pd.read_csv(
+            file_path,
+            header=0,
+            parse_dates=["pickup_datetime"],
+            infer_datetime_format=True
+        )
+        trip_dict["upload_date"] = upload_date
+        trip_dict.to_csv(
+            file_path,
+            header=True,
+            index=False
+        )
+
+    add_upload_date_1 = add_upload_date(
+        "/usr/local/airflow/include/data/yellow_tripdata_sample_2019-01.csv",
+        "{{ ds }}"
+    )
+    add_upload_date_2 = add_upload_date(
+        "/usr/local/airflow/include/data/yellow_tripdata_sample_2019-02.csv",
+        "{{ yesterday_ds }}"
+    )
+
+    @task
+    def upload_to_s3(file_path):
+        """
+        #### Upload task
+        Simply loads the file to a specified location in S3.
+        """
+        aws_configs = Variable.get("aws_configs", deserialize_json=True)
+        s3_bucket = aws_configs.get("s3_bucket")
+        s3_key = aws_configs.get("s3_key_prefix") + "/" + file_path
+        s3 = S3Hook()
+        s3.load_file(file_path, s3_key, bucket_name=s3_bucket, replace=True)
+        return {"s3_bucket": s3_bucket, "s3_key": s3_key}
```

Comment:
Is the returned `dict` being used anywhere? I may be missing it.

Reply:
It's a relic of when I copy/pasted from the Redshift DAG - that one does use the returned values.

Reply:
Ah OK. Let's 86 it here.

Reply:
It's gone with the use of `LocalFileSystemtoS3Operator` anyway.